File: C++/Daily002_easy/InterestCalculator.cpp
/** 
 * INTEREST CALCULATOR
 *
 * challenge #2 (easy)
 * http://www.reddit.com/r/dailyprogrammer/comments/pjbj8/easy_challenge_2/
 *
 * sarthak7u@gmail.com
 */

#include <iostream>
#include <math.h>

using namespace std;

double simpleInterest(double principle, double rate, double timePeriod);
double compoundInterest(double principle, double rate, double timePeriod);

int main()
{
	//variables for calculation
	double principle;
	double rate;
	double timePeriod;

	cout << "Enter the Principle amount: ";
	cin >> principle; 

	cout << "Enter the interest rate: ";
	cin >> rate;

	cout << "Enter the time period: ";
	cin >> timePeriod;

	//decides whether simple or compound interest
	int choice;
	cout << "Enter 0 for simple, 1 for compounded interest\n";
	cin >> choice;

	double interest;
	switch(choice)
	{
		case 0:
			interest = simpleInterest(principle, rate, timePeriod);
			cout << "The simple interst is: " << interest;
			break;
		case 1:
			interest = compoundInterest(principle, rate, timePeriod);
			cout << "The compounded interest is : "<< interest;
			break;

	}
	return 0;
}	

/**
 * computes simple interest
 */
double simpleInterest(double principle, double rate, double timePeriod)
{
	return principle * rate * timePeriod;
}

/**
 * computes compound interest
 */
double compoundInterest(double principle, double rate, double timePeriod)
{
	return (principle * pow((1 + rate), timePeriod)) - principle;
}

File: Python/Daily002_easy/InterestCalculator.py
## INTEREST CALCULATOR
##
## challenge #2 (easy)
## http://www.reddit.com/r/dailyprogrammer/comments/pjbj8/easy_challenge_2/
##
## sarthak7u@gmail.com

#computes simple interest
def simpleInterest(principle, rate, timePeriod):
	return principle * (rate/100) * timePeriod

#computes compound interest
def compoundInterest(principle, rate, timePeriod):
	return (principle * ((1 + (rate/100))**timePeriod)) - principle

principle = float(raw_input("Enter the Principle amount: "))
rate = float(raw_input("Enter the interest rate: "))
timePeriod = float(raw_input("Enter the time period: "))
choice = float(raw_input("Enter 0 for simple, 1 for compounded interest\n"))

if choice == 0:
	print "The simple interest is: ", simpleInterest(principle, rate, timePeriod)
elif choice == 1:
	print "The compound interest is: ", simpleInterest(principle, rate, timePeriod)